### PR TITLE
fix restoring iresearch view in cluster

### PR DESCRIPTION
**Pull request by graetzer**
Fixes https://github.com/arangodb/planning/issues/2826

http://jenkins01.arangodb.biz:8080/view/PR/job/arangodb-matrix-pr/633/

### CODE REVIEW

File: arangod/IResearch/IResearchLink.cpp [hunk 1]
```diff
@@ -214,19 +214,17 @@ bool IResearchLink::init(arangodb::velocypack::Slice const& definition) {
 
     auto logicalWiew = ci->getView(vocbase.name(), viewId);
     auto* wiew = LogicalView::cast<IResearchViewDBServer>(logicalWiew.get());
-
     if (wiew) {
-      auto collection = vocbase.lookupCollection(_collection.id());
-
+      // FIXME figure out elegant way of testing for cluster wide LogicalCollection
+      if (_collection.id() == _collection.planId() && _collection.isAStub()) {
```
Comment (gnusi) [inline arangod/IResearch/IResearchLink.cpp]: what `isAStub()` is intended for? I have never seen it before

Comment (gnusi) [inline arangod/IResearch/IResearchLink.cpp]: what `isAStub()` is intended for? I have never seen it before

**Comment by gnusi** [comment]
@graetzer I've just noticed that `LogicalCollection` wastes some space because of the negligent class layout (see sparsed bool members). That is minor issue, but could you please rearrange members in a proper way?

**Comment by graetzer** [comment]
Do you mean reordering the members by decreasing alignment ?

**Comment by gnusi** [comment]
@graetzer yes

**Comment by jsteemann** [comment]
There was one iresearch-related test failure in http://jenkins01.arangodb.biz:8080/job/arangodb-matrix-pr-linux/1521/EDITION=enterprise,STORAGE_ENGINE=rocksdb,TEST_SUITE=cluster,limit=linux&&test/artifact/testfailures.txt:
```
* Test "shell_server_aql"
    [FAILED]  js/common/tests/aql/aql-view-arangosearch-cluster.js

      "testInTokensFilterSortTFIDF" failed: at assertion #2: assertEqual: (other half) is not equal to (half)
(Error
    at assertEqual (/work/ArangoDB/js/common/modules/jsunity/jsunity.js:103:19)
    at Object.testInTokensFilterSortTFIDF (js/common/tests/aql/aql-view-arangosearch-cluster.js:250:7)
    at Object.run (/work/ArangoDB/js/common/modules/jsunity/jsunity.js:515:23)
    at Object.Run [as run] (/work/ArangoDB/js/common/modules/jsunity.js:171:24)
    at js/common/tests/aql/aql-view-arangosearch-cluster.js:502:9
    at js/common/tests/aql/aql-view-arangosearch-cluster.js:509:3
    at RunTest (/work/ArangoDB/js/common/modules/jsunity.js:229:12)
    at eval (eval at callback (/work/ArangoDB/js/actions/_admin/app.js:82:18), <anonymous>:2:52)
    at eval (eval at callback (/work/ArangoDB/js/actions/_admin/app.js:82:18), <anonymous>:2:136)
    at Function.callback (/work/ArangoDB/js/actions/_admin/app.js:82:18))


   Suites failed: 1 Tests Failed: 1
```

**Comment by KVS85** [comment]
Can you please merge devel once more and try to rerun PR tests? This particular test **testInTokensFilterSortTFIDF** could sporadically fail in cluster mode and fixing TFIDF/BM25 sorting is required by backporting newer code to arangodb from iresearch-toolkit. It is planned activity, but with less priority than others.
If it will fail again, please, just comment this test.

**Comment by graetzer** [comment]
http://jenkins01.arangodb.biz:8080/view/PR/job/arangodb-matrix-pr/645/

**Pull request merged**